modules/tinymce/src/core/main/ts/fmt/ListItemFormat.ts

```diff
@@ -32,25 +32,44 @@ const isRngEndAtEndOfElement = (rng: Range, elm: Element) => {
 
 const isEditableListItem = (dom: DOMUtils) => (elm: Element) => NodeType.isListItem(elm) && dom.isEditable(elm);
 
+const findParentListItem = (selection: EditorSelection, el: Element) => Optional.from(selection.dom.getParent(el, NodeType.isListItem));
+
 const getFullySelectedBlocks = (selection: EditorSelection) => {
   const blocks = selection.getSelectedBlocks();
   const rng = selection.getRng();
 
   if (selection.isCollapsed()) {
     return [];
-  } if (blocks.length === 1) {
+  }
+
+  // TINY-13197: The first block element returned by getSelectedBlocks() is not LI if the content is wrapped inside a block element, even when the content is fully selected. The following logic finds and replaces the first block with its parent LI
+  if (blocks.length > 0 && isRngStartAtStartOfElement(rng, blocks[0]) && !NodeType.isListItem(blocks[0])) {
+    findParentListItem(selection, blocks[0]).each((listItem) => {
+      blocks[0] = listItem;
```

Comment:
You're not checking if the block is the only child of the list item. What happens if there is a list with two paragraphs in it, and only the second is selected?

Reply:
> What happens if there is a list with two paragraphs in it, and only the second is selected?

I assume the list item style should not be updated, as observed with LI with content separated by `<br>`.

Reply:
What if both paragraphs are selected?

Reply:
If both paragraphs are selected it should update (because then all the content is selected).

I asked because this code looks like it will update the list item if only the second paragraph is selected. And I mean two `p`​ tags, not two lines separated by `br`​ tag. It's a rare case, but I've seen weirder.

Reply:
`isRngStartAtStartOfElement` check ensures that the list item is updated only when all content is selected.

Generally, I replicate how the formatting works for text content (without block element). Hopefully it is good enough

Reply:
The test name: `TINY-13197: Should only apply color to the selected blocks and not update the first LI as it is not fully selected`